### PR TITLE
[REF] Convert CRM_Core_Error::fatal to CRM_Core_Error::statusBounce i…

### PR DESCRIPTION
**Pull request by seamuslee001**
…n Extension folder

Overview
----------------------------------------
This converts from deprecated function `CRM_Core_Error::fatal` to using `CRM_Core_Error::statusBounce`

Before
----------------------------------------
Deprecated function used in the Extensions subsystem

After
----------------------------------------
Non depreacted function used

ping @yashodha @eileenmcnaughton

### CODE REVIEW

File: CRM/Extension/Downloader.php [hunk 1]
```diff
@@ -105,7 +105,7 @@ public function download($key, $downloadUrl) {
     $destDir = $this->containerDir . DIRECTORY_SEPARATOR . $key;
 
     if (!$downloadUrl) {
-      CRM_Core_Error::fatal(ts('Cannot install this extension - downloadUrl is not set!'));
+      CRM_Core_Error::statusBounce(ts('Cannot install this extension - downloadUrl is not set!'));
```
Comment (eileenmcnaughton) [inline CRM/Extension/Downloader.php]: is this the form - otherwise an exception makes sense

**Comment by eileenmcnaughton** [comment]
@seamuslee001 were you going to update this to throw exceptions?

**Comment by seamuslee001** [comment]
done @eileenmcnaughton

File: CRM/Extension/Downloader.php [hunk 2]
```diff
@@ -105,7 +105,7 @@ public function download($key, $downloadUrl) {
     $destDir = $this->containerDir . DIRECTORY_SEPARATOR . $key;
 
     if (!$downloadUrl) {
-      CRM_Core_Error::fatal(ts('Cannot install this extension - downloadUrl is not set!'));
+      throw new CRM_Extension_Exception(ts('Cannot install this extension - downloadUrl is not set!'));
```
Comment (eileenmcnaughton) [inline CRM/Extension/Downloader.php]: ohh flash

**Pull request merged**